### PR TITLE
Don't compile the Skia GL backend on iOS.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Depends on https://skia-review.googlesource.com/c/skia/+/562341

### CODE REVIEW

**Comment by chinmaygarde** [comment]
It isn't clear what needs to be done to land the Skia patch in lockstep with this one. Also, the !OS_FUCHSIA stuff can probably be avoided but decided I could do that in another patch to keep the differences small as I couldn't test the Fuchsia stuff on Mac.

File: shell/common/shell_io_manager.cc
Comment (zanderso) [inline shell/common/shell_io_manager.cc:16]: Is `SK_GL` any different from `SHELL_ENABLE_GL` in this context?

File: tools/gn
Comment (zanderso) [inline tools/gn:230]: There's similar code starting around line 407.

File: shell/common/shell_io_manager.cc
Comment (chinmaygarde) [reply shell/common/shell_io_manager.cc:16]: Yes, `SHELL_ENABLE_GL` is only available in the GPU specific code. Unfortunately, this is in `shell/common`. Ideally, this selection should be moved to `shell/gpu` but that is a larger change.

File: tools/gn
Comment (chinmaygarde) [reply tools/gn:230]: Done.

**Review by zanderso — APPROVED** [review]

File: shell/common/shell_io_manager.cc
Comment (dnfield) [reply shell/common/shell_io_manager.cc:16]: nit: add a TODO and file a bug to move this to shell/gpu

**Comment by zanderso** [comment]
Presumably the Skia roll with the above CL will fail, but will pass when patched into this PR. That's probably how we'll have to land them.

**Review by dnfield — APPROVED** [review]

**Comment by chinmaygarde** [comment]
Roger. That doesn't have a review yet. Once that lands and a roll is attempted, I'll rebase.

**Comment by chinmaygarde** [comment]
The Skia roller is blocked on an unrelated issue and I am waiting for the roll with the linked CL to be attempted.

**Comment by chinmaygarde** [comment]
The patch has rolled in and I've rebased here https://github.com/flutter/engine/pull/34949.

File: shell/common/shell_io_manager.cc
Comment (chinmaygarde) [reply shell/common/shell_io_manager.cc:16]: https://github.com/flutter/engine/pull/34959

**Pull request merged**
